api/server/router/network/network_routes.go

```diff
@@ -70,14 +72,62 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 		return err
 	}
 
-	nw, err := n.backend.FindNetwork(vars["id"])
-	if err != nil {
-		if nr, err := n.cluster.GetNetwork(vars["id"]); err == nil {
-			return httputils.WriteJSON(w, http.StatusOK, nr)
+	// In case multiple networks have duplicate names, return error.
+	// TODO (yongtang): should we wrap with version here for backward compatibility?
+
+	// Find based on full ID
+	nr, _ := n.cluster.GetNetworks()
+	for _, network := range nr {
+		if network.ID == vars["id"] {
+			return httputils.WriteJSON(w, http.StatusOK, network)
+		}
+	}
+	nw := n.backend.GetNetworks()
+	for _, network := range nw {
+		if network.ID() == vars["id"] {
+			return httputils.WriteJSON(w, http.StatusOK, n.buildNetworkResource(network))
+		}
+	}
+
+	list := []types.NetworkResource{}
+
+	// Find based on full name, reutrns true only if no duplicates
+	for _, network := range nr {
+		if network.Name == vars["id"] {
+			list = append(list, network)
+		}
+	}
+	for _, network := range nw {
+		if network.Name() == vars["id"] {
+			list = append(list, *n.buildNetworkResource(network))
+		}
+	}
+	if len(list) == 1 {
+		return httputils.WriteJSON(w, http.StatusOK, list[0])
+	}
+	if len(list) > 1 {
+		return fmt.Errorf("network %s is ambiguous (%d matches found based on name)", vars["id"], len(list))
+	}
+
+	// Find based on partial ID, returns true only if no duplicates
```

Comment:
Comment says partial ID, but the code appears to be looking at `Name()` in the second loop.